numpy/typing/tests/test_typing.py

```diff
@@ -84,7 +84,7 @@ def run_mypy() -> None:
     """
     if (
         os.path.isdir(CACHE_DIR)
-        and bool(os.environ.get("NUMPY_TYPING_TEST_CLEAR_CACHE", True))
+        and bool(os.environ.get("NUMPY_TYPING_TEST_CLEAR_CACHE", "True"))
```

Comment:
it's technically still `True`, but I don't see how this is better tbh 🤷🏻

Reply:
The general rule is that `os.environ.get` should always return `str` (if the environment variable is set) or `None` (if the environment variable is not set).

This specific case is a clever shorthand for the horribly long:
```
and (bool(os.environ.get("NUMPY_TYPING_TEST_CLEAR_CACHE"))
     if os.environ.get("NUMPY_TYPING_TEST_CLEAR_CACHE") is not None else True)
```
It's probably best to stick with `# noqa: PLW1508`!

Reply:
TBH I wasn't even aware of this env var, and was cleaning `.mypy_cache` by hand haha. But yea, a `# noqa` seems justified here.